### ISSUE TITLE
How to get rule ID in a formatter?

### ISSUE THREAD

**Issue by steelbrain** [eslint/eslint#4394]
I am using eslint `v1.9.0` and don't get `ruleId` in my formatter. These are all the keys that I get `['column', 'fatal', 'line', 'message', 'severity']`. I am getting these using `Object.keys(results[0].messages[0])` with a file having this as content
```js
var a = 1."";
    as

```

Am I doing anything wrong or is this actually a bug?

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by steelbrain** [comment]
@eslintbot, well, I already did that :wink:

**Comment by nzakas** [comment]
`fatal` errors don't have a rule ID because they come from the parser. Our Node.js API docs go into this a bit: http://eslint.org/docs/developer-guide/nodejs-api.html#linter

**Comment by steelbrain** [comment]
@nzakas That works! Thanks!
